### PR TITLE
Clamp API sentinel temperatures to physical device limits

### LINKED ISSUE #50: Bug: API sentinel values (999, -1) leak into HA as raw temperatures (causing extreme temperature spikes and dips in Fahrenheit)
*Disclaimer: I used an AI assistant to help me troubleshoot the logs and write this bug report.*

**Describe the bug**
The Sleep.me developer API uses sentinel values (`999` for Max Heat and `-1` for Max Cool). The integration is currently passing these raw values directly to Home Assistant. 
* If Home Assistant is set to metric, the UI displays `-1°C` and `999°C`. 
* If Home Assistant is set to Fahrenheit, it applies the conversion formula, causing the dashboard and history logs to show absurd target temperatures of `30.2°F` and `1830.2°F`.

**Hardware & Environment**
* **Hardware:** Two Chilipad Dock Pro units (original; version 1), fully updated.
* **Specs:** The normal temperature control range for these units is 13°C to 48°C (55.4°F to 118.4°F), plus a 'Max Cool' and 'Max Heat' preset from the API.
* **Reproducibility:** I can reproduce these extreme temperature readings on both of my docks.

**Expected behavior**
The integration should intercept the `999` and `-1` API placeholders and map them to the unit's actual limits (`48.0` and `13.0` Celsius) before passing them to Home Assistant. Additionally, when a user sets the dashboard slider to max heat/cool, those physical limits should be mapped back to `999`/`-1` before the payload is sent to the API.

**Steps to Reproduce**
1. Set the Dock Pro to the "Max Heat" or "Max Cool" limit (e.g., using a Sleep.me automation or preset).
2. Observe the target temperature display as `999°C` / `-1°C` (in metric) or spike/dip to `1830°F` / `30°F` (in Fahrenheit).

### PR DESCRIPTION
**Pull request by derekcentrico**
## Summary

- The SleepMe API uses sentinel values (`-1` for MAX COLD, `999` for MAX HEAT) in `set_temperature_c`. The integration was passing these to HA as real Celsius temperatures, which HA then converted to Fahrenheit, producing 30.2°F and 1830.2°F on the dashboard and in history graphs.
- `target_temperature` now clamps sentinels to the device's actual physical limits (13.0°C / 48.0°C) before exposing them to HA. Preset mode detection and outbound API calls still use the raw sentinel values.
- Also fixed a subtle edge case where switching directly between presets (e.g., Max Heat to Max Cool) would overwrite the user's previously saved temperature with the clamped sentinel value.

Fixes #50

## Test plan

- [x] All 69 tests pass (8 new: 4 for sentinel clamping in climate, 4 for `clamp_api_sentinel` helper)
- [ ] Set device to Max Heat via the SleepMe app, verify HA shows 48.0°C (118.4°F) instead of 999°C (1830.2°F)
- [ ] Set device to Max Cool, verify HA shows 13.0°C (55.4°F) instead of -1°C (30.2°F)
- [ ] Verify preset mode indicator still correctly shows "Max Heat" / "Max Cool"
- [ ] Set a normal temp (e.g. 24°C), switch to Max Heat, then Max Cool, then "none", verify it restores 24°C

---

@rsampayo Would appreciate your thoughts on this approach. The clamping happens at the presentation layer only, so sentinel values still flow through to the API on outbound calls. Let me know if you'd prefer a different spot in the data path for this mapping.

## Summary by CodeRabbit

- **Bug Fixes**
  - Extreme cooling and heating preset values now display as valid physical temperature limits instead of API sentinel values.
  - Preset switching preserves explicitly selected temperatures when returning to normal mode.
  - Standard temperature values continue to pass through unchanged.

- **Tests**
  - Added coverage for extreme preset handling, normal temperatures, preserved setpoints, and empty values.

### CODE REVIEW

File: custom_components/sleepme_thermostat/climate.py [hunk 1]
```diff
@@ -136,10 +136,8 @@ def current_temperature(self) -> float | None:
 
     @property
     def target_temperature(self) -> float | None:
-        """Optimistic target wins until the server confirms or the window expires."""
-        server_value = self.coordinator.data["control"].get("set_temperature_c")
-        optimistic = self._effective_optimistic_temp(server_value)
-        return optimistic if optimistic is not None else server_value
+        """Return the target temperature, clamping API sentinels to physical limits."""
+        return clamp_api_sentinel(self._raw_setpoint())
```
[the remark this answers is not in the document]
Comment (derekcentrico) [reply custom_components/sleepme_thermostat/climate.py:140]: Setting 48.0°C and MAX HEAT (999) are distinct operations per the API spec. The slider sends real temperatures within the 13.0–48.0 range; the preset buttons send sentinels via `PRESET_TEMPERATURES`. A user who drags to 48°C wants "set to 48 degrees," not "activate max heat mode." The outbound preset path already sends the correct sentinel values and is covered by `test_preset_mode_max_cool`.

**Comment by rsampayo** [comment]
@derekcentrico Sorry for the slow reply here, and thank you for picking this up, shipping it, and following up with the reporter on #50.

I've gone through the change and I'm happy with it:

- Clamping at the presentation layer is the right spot. Sentinels still flow outbound and `preset_mode` still derives from the raw setpoint, which matches the API spec (`-1` / `999` are distinct operations from a real 13.0 / 48.0 request). I agree with your call on the CodeRabbit thread.
- The `_raw_setpoint()` split cleanly fixes the preset → preset restore bug too.
- Ran the suite locally on HA 2026.5.2: 69 passed, 89.78% coverage. ruff / mypy / the HA matrix were green in CI.

One small thing before this goes stable: `manifest.json` still says `4.2.0` under the `v4.2.1-rc1` tag (every prior release bumped it in the same commit). Let's bump it to `4.2.1` when we promote the rc once @damienbibeau confirms the fix.

Thanks again.
